trieste/acquisition/function/greedy_batch.py

```diff
@@ -351,3 +355,286 @@ def __call__(self, x: TensorType) -> TensorType:
         p = -5  # following experiments of :cite:`Alvi:2019`.
         penalization = ((pairwise_distances / (self._radius + self._scale)) ** p + 1) ** (1 / p)
         return tf.reduce_prod(penalization, axis=-1)
+
+
+class Fantasizer(GreedyAcquisitionFunctionBuilder[ProbabilisticModel]):
```

Comment:
can you please raise an issue about adding integration tests in a future PR so we don't forget about it